src/lumigo_tracer/lambda_tracer/lambda_reporter.py

```diff
@@ -143,16 +145,17 @@ def report_json(
     get_logger().info(f"reporting the messages: {msgs[:10]}")
     try:
         prune_trace: bool = not os.environ.get("LUMIGO_PRUNE_TRACE_OFF", "").lower() == "true"
-        to_send = _create_request_body(msgs, prune_trace).encode()
+        should_try_zip: bool = _should_try_zip()
+        to_send: Union[str, List[str]] = _create_request_body(msgs, prune_trace, should_try_zip)
     except Exception as e:
         get_logger().exception("Failed to create request: A span was lost.", exc_info=e)
         return 0
     if should_use_tracer_extension():
         with lumigo_safe_execute("report json file: writing spans to file"):
             write_spans_to_files(spans=msgs, is_start_span=is_start_span)
         return 0
-    if region == CHINA_REGION:
-        return _publish_spans_to_kinesis(to_send, CHINA_REGION)
+    if region == CHINA_REGION and isinstance(to_send, str):
+        return _publish_spans_to_kinesis(to_send.encode(), CHINA_REGION)
```

Comment:
Did you test sending zipped spans in this region?

Reply:
@sagivoululumigo I have it changed today + added unit test to handle.
we do not want to do Zipped spans for China region

Reply:
yes, I have unit test for this